### DISCUSSION TITLE
Trouble uploading code

### DISCUSSION THREAD

**Discussion by Buergels** [NicoHood/HoodLoader2#128] [Q&A]
Hey, just found out about hoodloader2 and I am interested in using it with a Mega. Got everything working on the board and I believe in the IDE as well but I keep getting an error message when I try to compile code.

cc1.exe: error: -fno-fat-lto-objects are supported only with linker plugin.
exit status 1
Compilation error: exit status 1

I recon something is messed up with the IDE, everything is up to date.
Any help is appreciated and thanks in advance.

**Comment by Abbence** [comment]
Hi!
I have an issue very similar to yours. I'm new to arduino, I use an Uno and I want to make a HID project using HL2 and the 16u2.
If I include the "HID-Project.h", I get the error:
`C:\Users\[me]\AppData\Local\Arduino15\packages\arduino\tools\avr-gcc\4.8.1-arduino5/bin/avr-gcc-ar: Cannot find plugin 'liblto_plugin-0.dll'`
If I don't include the library, then I get the compilation error you also get.
Searched for a while on google but I don't seem to understand what the problem is and how should I solve it.

**Comment by amonreal0** [comment]
Hi

I was having the same issue and I found a closed issue reporting similar behavior. https://github.com/NicoHood/HoodLoader2/issues/124
I believe HoodLoader2 is not compatible with Arduino IDE 2.x.x so you have to use the legacy IDE
Hope that helps
